cloud/services/compute/firewalls/reconcile.go

```diff
@@ -28,8 +28,8 @@ import (
 // Reconcile reconcile cluster firewall compoenents.
 func (s *Service) Reconcile(ctx context.Context) error {
 	log := log.FromContext(ctx)
-	if s.scope.IsSharedVpc() {
-		log.V(2).Info("Shared VPC enabled. Ignore Reconciling firewall resources")
+	if s.scope.SkipFirewallRuleCreation() {
```

Comment:
BTW I like this a lot, even if SkipFirewallRuleCreation just called IsSharedVpc, it is much more self-documenting to have self-descriptive functions like this and put the logic in those functions, rather than put the logic in the caller.